controllers/virtualmachine/virtualmachine_controller.go

```diff
@@ -156,6 +156,24 @@ func classToVMMapperFn(ctx *pkgctx.ControllerManagerContext, c client.Client, is
 	}
 }
 
+func upgradeSchema(ctx *pkgctx.VirtualMachineContext) {
+	// If empty, this VM was created before v1alpha3 added the spec.biosUUID field.
+	if ctx.VM.Spec.BiosUUID == "" {
+		ctx.VM.Spec.BiosUUID = ctx.VM.Status.BiosUUID
+		ctx.Logger.Info("Upgrade VirtualMachine biosUUID")
+	}
+
+	// If empty, this VM was created before v1alpha3 added the instanceID field
+	if bs := ctx.VM.Spec.Bootstrap; bs != nil {
+		if ci := bs.CloudInit; ci != nil {
+			if ci.InstanceID == "" {
+				ci.InstanceID = string(ctx.VM.UID)
+				ctx.Logger.Info("Upgrade VirtualMachine instanceID")
```

Comment:
Please add more details on logging. eg,`Update VirtualMachine cloudInit instanceID to use biosUUID`

Reply:
Added field path and value to both logs